packages/expo-cli/src/commands/client/index.js

```diff
@@ -19,15 +23,39 @@ const { IOS } = PLATFORMS;
 
 export default program => {
   program
-    .command('client:ios')
+    .command('client:ios [project-dir]')
     .option(
       '--apple-id <login>',
       'Apple ID username (please also set the Apple ID password as EXPO_APPLE_PASSWORD environment variable).'
     )
     .description(
       'Build a custom version of the Expo Client for iOS using your own Apple credentials and install it on your mobile device using Safari.'
     )
-    .asyncAction(async options => {
+    .asyncActionProjectDir(async (projectDir, options) => {
+      const servicesDisabled = {
+        pushNotifications:
+          'not yet available until API tokens are supported for the Push Notification system',
+      };
+
+      // get custom project manifest if it exists
+      // Note: this is some random user's project, NOT the expo client manifest
+      const spinner = ora(`Finding custom configuration for the Expo client...`).start();
+      const appJsonPath = options.config || path.join(projectDir, 'app.json');
+      const appJsonExists = await ConfigUtils.fileExistsAsync(appJsonPath);
+      const { exp } = appJsonExists ? await ConfigUtils.readConfigJsonAsync(projectDir) : {};
+
+      if (exp) {
+        spinner.succeed(`Found custom configuration for the Expo client at ${appJsonPath}`);
+      } else {
+        spinner.warn(`Unable to find custom configuration for the Expo client.`);
+      }
+      if (!_.has(exp, _ => _.ios.config.googleMapsApiKey)) {
```

Comment:
```suggestion
      if (!_.has(exp, 'ios.config.googleMapsApiKey')) {
```

Reply:
![Screenshot 2019-06-26 at 09 47 26](https://user-images.githubusercontent.com/5256730/60161264-778daa80-97f7-11e9-8f52-b43b51470206.png)
;)

Reply:
```suggestion
      if (!idx(exp, _ => _.ios.config.googleMapsApiKey)) {
```
(I think this can be better typechecked by TypeScript compiler than the stringy `_.has` syntax.)

Reply:
@fson @dsokal we should decide if we want to prefer `idx` or `_.has`. I'm just going to keep `_.has` for now because that's what I've been changing it to across different pr's for consistency sake.

Reply:
In my opinion, the choice is obvious. It's better to have one npm package and not two.